### ISSUE TITLE
Some questions about datasets

### ISSUE THREAD

**Issue by Ru1zhi** [Wanggcong/SparseNeRF#7]
Thanks for the open source behavior. I have a a few questions about the data:
1. [here](https://github.com/Wanggcong/SparseNeRF/blob/cad362cef9c62dd2659cd5342e10dab132b61795/internal/datasets_depth_llff_dtu.py#L518-L537), you set different sampling windows for different scenes instead of sampling randomly on the image, what is the purpose of doing this?
2. It seems wrong [here](https://github.com/Wanggcong/SparseNeRF/blob/cad362cef9c62dd2659cd5342e10dab132b61795/internal/datasets_depth_llff_dtu.py#L1065), random_poses.append() should be inside the loop?
3. The default settings in the code all use the regularization constraints for randompose, so are the metrics reported in the paper based on that?

**Comment by Wanggcong** [comment]
Thank you for your comment. 

Q1. Yes. The reason is that even the depth ranking of the pre-trained depth estimation is not that accurate. This is why I claimed local depth ranking in the paper.

Q2. I also observed it, which is from [RegNeRF](https://github.com/google-research/google-research/tree/master/regnerf). I found that RegNeRF only use one random poses for LLFF and many random poses for DTU. You might check it in the RegNeRF. I think the reason behind this is that the evaluation of LLFF only cares about front-forward scenes, which only has limited field of view. Even we use more random poses, the metric is about limited field of views. I have evaluated this before. For LLFF, even I changed it inside the loop, the evaluation results do not improve. For DTU, it may work.

Q3. Yes.

Please let me know if I make it clear.

**Comment by Ru1zhi** [comment]
The window size setting varies a lot for different scenes, what is this window size setting based on? Does the size of the local window have a major impact on performance?

**Comment by Wanggcong** [comment]
It is based on 1) the object size and 2)the accuracy of pre-trained depth estimation models. 

- For example, for DTU, the object size is about half of the image size. In this case, we might select a large window size. If the scene contains lots of objects, the local window size does not have huge impact on the performance. 

- Ideally, if the pre-trained depth estimation model is perfect, it is suggested to use global sampling on the image. 

How to select the window size? Roughly, you might the window size as the object size. If the scenes contains lots of objects with different sizes, we might select a small size, say 30. But if you could slight tune this size since it could be hard to determine the accuracy of pre-trained depth estimation.

**Comment by Ru1zhi** [comment]
Very clear explanation, thanks again!
